### PR TITLE
reduce scope of variables, minor fixes

### PR DESCRIPTION
**Pull request by rabauke**
Fixes for real bugs are indicated below.

### CODE REVIEW

File: src/common/variables.c [hunk 1]
```diff
@@ -513,7 +513,7 @@ static char *variable_get_value(dt_variables_params_t *params, char **variable)
           if(length >= 0)
             end = g_utf8_offset_to_pointer(start, MIN(length, start_utf8_length));
           else
-            end = g_utf8_offset_to_pointer(base_value + base_value_length, MAX(length, -1 * start_utf8_length));
```
Comment (rabauke) [inline src/common/variables.c]: The result of the multiplication of -1 by the unsigned `start_utf8_length` and `base_value_utf8_length` yields an unsigned integer. Signed variables must be used here.

File: src/common/variables.c [hunk 2]
```diff
@@ -454,7 +454,7 @@ static char *variable_get_value(dt_variables_params_t *params, char **variable)
       */
       {
         char *replacement = expand(params, variable, ')');
-        if(!base_value || !*base_value)
```
Comment (rabauke) [inline src/common/variables.c]: This would lead to a null pointer dereference if `base_value` would be null.

File: src/common/tags.c [hunk 3]
```diff
@@ -1323,7 +1323,7 @@ static gchar *dt_cleanup_synonyms(gchar *synonyms_entry)
         e--;
       }
       e = *entry;
-      while(*e == ' ' && *e != '\0') e++;
+      while(*e == ' ') e++;
       if(*e)
```
Comment (rabauke) [inline src/common/tags.c]: Second test is redundant.

File: src/common/tags.c [hunk 4]
```diff
@@ -554,7 +554,7 @@ void dt_tag_attach_string_list(const gchar *tags, gint imgid, const gboolean und
           e--;
         }
         e = *entry;
-        while(*e == ' ' && *e != '\0') e++;
```
Comment (rabauke) [inline src/common/tags.c]: Second test is redundant.

File: src/common/opencl.c [hunk 5]
```diff
@@ -126,7 +126,7 @@ int dt_opencl_get_device_info(dt_opencl_t *cl, cl_device_id device, cl_device_in
 error:
   free(*param_value);
   *param_value = NULL;
-  param_value_size = 0;
+  *param_value_size = 0;
   return err;
```
Comment (rabauke) [inline src/common/opencl.c]: Setting the _pointer_ `param_value_size` (a  function parameter, i.e., a local variable) to 0 makes no sense here.

Comment (TurboGit) [reply src/common/opencl.c]: Indeed!

File: src/common/variables.c (hunk 2 above)
Comment (TurboGit) [reply src/common/variables.c]: No as in C operator || does not evaluate the second argument if first one is true. Here NULL will make the first argument true and so if won't evaluate second one.

File: src/common/variables.c (hunk 1 above)
Comment (TurboGit) [reply src/common/variables.c]: Right!

File: src/common/variables.c (hunk 2 above)
Comment (rabauke) [reply src/common/variables.c]: You are right here. Nevertheless, the original code was confusing in my view. The null pointer test can be omitted.`!base_value ` _always_ evaluates to false, see comment in line 442. I suggest to omit the test here as done in line 473.

Comment (TurboGit) [reply src/common/variables.c]: Agreed, the test should just be:

`if(!*base_value)`

**Comment by TurboGit** [comment]
All good to me now. Thanks.

**Pull request merged**